### PR TITLE
Add a second check to find employees by username.

### PR DESCRIPTION
**Pull request by brenns10**
I initially set up the GAE app so that it was open to any Google user, not just `@case.edu` domain users. This was out of ignorance -- I didn't know the setting was available. This was not an issue because the import script ensures that only `case.edu` domain users get `Employee` entries, and thus when outside users access the site, an `EmployeeNotFound` exception is raised and they see an internal error. However, this doesn't happen when viewing the leaderboard and explore pages, so lots of love can be leaked.

When I changed the setting to restrict to `@case.edu` domain users, the `User` objects from GAE changed somehow, so that the Employee lookup query failed for *every* user.

This pull request addresses this query error.  If the lookup by `User` object fails, we parse the email and get the username, and then we lookup the Employee by username.

This is already running successfully in prod, but I may modify it a bit before merging. I may want to save the new version of the User object on the Employee so that subsequent queries don't hit this logic.

@rockdog notified me about this issue and may be interested in this fix.

### CODE REVIEW

**Comment by brenns10** [comment]
This fix itself may not be upstreamable, since it's a fix that's very specific to our issue, but maybe there should be a note in the README so that people know to switch their project to domain-only *before* importing employees.

File: models/employee.py [hunk 1]
```diff
@@ -46,6 +47,10 @@ def get_current_employee(cls):
         user_email = user.email()
         employee = cls.query(cls.user == user, cls.terminated == False).get()  # noqa
         if employee is None:
+            # if you change the GAE domain settings, the user objects change
+            # and the above query fails. this query will work for that
+            employee = cls.get_employee_by_email(user_email)
+        if employee is None:
```
Comment (mtbentley) [inline models/employee.py:53]: I'm not a huge fan of doing `if employee is None` twice, but idk if there's a better way to do it.

Comment (brenns10) [reply models/employee.py:53]: I don't think there is a better way to do it.

Comment (brenns10) [reply models/employee.py:53]: I mean I guess `employee = employee or cls.get_employee_by_email(user_email)`.

But that's basically the same thing.

Comment (mtbentley) [reply models/employee.py:53]: Yeah I suppose it's OK as is.  I just did a bit of a double take when I read it.

**Pull request merged**